lib/scope.js

```diff
@@ -991,13 +994,78 @@ module.exports = {
 
         await scope.afterStep(scope, { waitForShowIf: true });
         
+      } else if ( type === `file` ) {
+        await scope.uploadFiles( scope, { handle, set_to });
+        await scope.afterStep(scope, { waitForShowIf: true });
+
       } else {
         await scope.setText( scope, { handle, set_to });
-        await scope.afterStep(scope );  // No showifs for this one?
+        await scope.afterStep(scope, { waitForShowIf: true });
       }
     },  // Ends scope.setField.input()
   },
 
+  uploadFiles: async function ( scope, { handle, set_to }) {
+    /* Try to upload one or more files to a file-type input field. */
+
+    // Allow a typo where someone forgot to include a space after a comma?
+    let file_names = set_to.split(`,`);
+    
+    // Tests could be in one of two places. Figure out which directory they're in
+    // Makes an absolute path from the directory in which `npm run` was used plus our internal folders
+    let dir_source = `${ process.cwd() }/docassemble/${ env_vars.REPO_URL.match(/docassemble-(.*)/)[1] }/data/sources`;
+    let dir_features = `${ process.cwd() }/tests/features`;
+    let dir = null;
+
+    // Make sure one of the valid directories exist.
+    try {
+      // Try the most common directory first
+      fs.accessSync( dir_source );
+      dir = dir_source;
+      // If they have both directories, let them know which directory will be used.
+      try {
+        fs.accessSync( dir_features );
+        await scope.addToReport( scope, { type: `warning`, value: `WARNING: You have both a docassemble package 'sources' folder and a 'test/features' folder. These tests will use the package's 'sources' folder.` });
+      } catch ( error_features ) { `all's well, do nothing`; }
+
+    } catch ( error_source ) {
+      try {
+        fs.accessSync( dir_features );
+        dir = dir_features;
+      } catch ( error_features ) {
+        await scope.addToReport( scope, { type: `warning`, value: `WARNING: There are two places where you can store files to upload: "${ dir_source }" and "${ dir_features }". These tests could not find either of those.` });
+        throw error_source;  // Most people will want to know the sources folder path;
+      }
+    }
+
+    let paths = [];
+    // If a file exist, add it to the list of files to be uploaded. If it doesn't, add a warning.
+    for ( let name of file_names ) {
+
+      // If the developer correctly used a ", ", there will be a space at the
+      // start of the string. Remove it. If they forgot the space, that's fine too.
+      name = name.replace(/^\s/, '');
+      let file_path = path.join( dir, name );
```

Comment:
Looks like we're enforcing that all files listed in one line must be in the same directory (i.e., you can't have `some_png_1` in sources and `some_png_2` in features. We should document this.

Reply:
Great idea, thanks! #347

Reply:
Now see https://github.com/SuffolkLITLab/docassemble-AssemblyLine-documentation/issues/207